libraries/cmd/armed.cpp

```diff
@@ -198,10 +198,14 @@ void Armed::ActivateGuns(const WeaponInfo *sz, bool ms) {
 }
 
 void Armed::Fire(unsigned int weapon_type_bitmask, bool listen_to_owner) {
-    Unit *unit = static_cast<Unit *>(this);
+    Unit *unit = vega_dynamic_cast_ptr<Unit>(this);
 
-    if ((unit->cloak.Active() && !configuration()->weapons.can_fire_in_cloak) ||
-            (unit->ftl_drive.Enabled() && !configuration()->weapons.can_fire_in_spec)) {
+    if (unit->cloak.Active() && !configuration()->weapons.can_fire_in_cloak) {
+        VS_LOG(debug, (boost::format("%1%: can't fire while cloaked") % __FUNCTION__));
+        UnFire();
+        return;
+    } else if (unit->ftl_drive.Enabled() && !configuration()->weapons.can_fire_in_spec) {
```

Comment:
Isn't the else redundant here?

Reply:
I suppose it could be just another `if`, after the previous one. I don't see that it makes much difference, personally.

Reply:
It doesn't.